src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBResource.cs

```diff
@@ -14,15 +16,27 @@ public class AzureCosmosDBResource(string name, Action<AzureResourceInfrastructu
     IResourceWithConnectionString,
     IResourceWithEndpoints
 {
-    internal List<string> Databases { get; } = [];
+    internal List<CosmosDBDatabase> Databases { get; } = [];
 
     internal EndpointReference EmulatorEndpoint => new(this, "emulator");
 
     /// <summary>
     /// Gets the "connectionString" reference from the secret outputs of the Azure Cosmos DB resource.
+    ///
+    /// This is used when Entra ID authentication is used. The connection string is an output of the bicep template.
     /// </summary>
     public BicepSecretOutputReference ConnectionString => new("connectionString", this);
```

Comment:
Isn't this supposed to be:

```suggestion
    public BicepOutputReference ConnectionString => new("connectionString", this);
```

It's not a secret reference anymore - it is just a normal bicep output.

Reply:
That would be a breaking change. How do you suggest we deal with it?

Reply:
This is why I opt to not make things public unless we absolutely need to make them public.

I would suggest just making the breaking change, and document it. The fact that we are switching to managed identity by default is going to be an even bigger breaking change.

Reply:
Actually, we can just expose another property

Reply:
This shouldn't be obsolete right?

Reply:
I wanted to get some feedback on it. So just add a new property without obsoleting the previous one? Just to have a better type? I personally prefer to obsolete but I may not think about some implications.

Reply:
This is still valid in the access key mode.

Reply:
OK, now I understand